src/SprykerEco/Zed/Adyen/Persistence/AdyenEntityManager.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Adyen\Persistence;
+
+use Generated\Shared\Transfer\PaymentAdyenApiLogTransfer;
+use Generated\Shared\Transfer\PaymentAdyenOrderItemTransfer;
+use Generated\Shared\Transfer\PaymentAdyenTransfer;
+use Generated\Shared\Transfer\SpyPaymentAdyenApiLogEntityTransfer;
+use Generated\Shared\Transfer\SpyPaymentAdyenEntityTransfer;
+use Generated\Shared\Transfer\SpyPaymentAdyenOrderItemEntityTransfer;
+use Spryker\Zed\Kernel\Persistence\AbstractEntityManager;
+use SprykerEco\Zed\Adyen\Persistence\Mapper\AdyenPersistenceMapperInterface;
+
+/**
+ * @method \SprykerEco\Zed\Adyen\Persistence\AdyenPersistenceFactory getFactory()
+ */
+class AdyenEntityManager extends AbstractEntityManager implements AdyenEntityManagerInterface
+{
+    /**
+     * @param \Generated\Shared\Transfer\PaymentAdyenTransfer $paymentAdyenTransfer
+     *
+     * @return \Generated\Shared\Transfer\PaymentAdyenTransfer
+     */
+    public function savePaymentAdyen(PaymentAdyenTransfer $paymentAdyenTransfer): PaymentAdyenTransfer
+    {
+        $entityTransfer = $this
+            ->getMapper()
+            ->mapPaymentAdyenTransferToEntityTransfer(
+                $paymentAdyenTransfer,
+                new SpyPaymentAdyenEntityTransfer()
```

Comment:
Why do you need to create it here? Maybe create dit in the mapper and return?

Reply:
I think it's not really matter where to create a transfer, but in this version I cover case when I need to update existing transfer.